src/codec/rtu.rs

```diff
@@ -540,16 +541,18 @@ mod tests {
         #[test]
         fn decode_partly_received_client_message() {
             let mut codec = ClientCodec::default();
-            let mut buf = BytesMut::from(vec![
-                0x12, // slave address
-                0x02, // function code
-                0x03, // byte count
-                0x00, // data
-                0x00, // data
-                0x00, // data
-                0x00, // CRC first byte
-                      // missing crc second byte
-            ]);
+            let mut buf = BytesMut::from(
+                &[
```

Comment:
What was the motivation to replace vec! with an array coerced to a slice? Now BytesMut::from will allocate the Vec internally: https://docs.rs/bytes/0.5.3/src/bytes/bytes_mut.rs.html#1013

Doesn't really matter, I'm just asking out of curiosity.

Reply:
I dont actually have an opinion here - `impl From<Vec<u8>>` was removed and I did the simplest thing to fix it. Maybe `from_vec` would be better here as a more explicit alternative.

Reply:
My mistake, I forgot to actually build the tests :see_no_evil: